presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1911,7 +1911,9 @@ public ConnectorPushdownFilterResult pushdownFilter(ConnectorSession session, Co
         domainPredicate.getDomains().get().keySet().stream()
                 .map(Subfield::getRootName)
                 .forEach(predicateColumnNames::add);
-        extractAll(decomposedFilter.getRemainingExpression()).stream()
+        // optimize the remaining expression before adding to predicateColumns
```

Comment:
Let's add a test case for expression over multiple columns being optimized into an expression over fewer columns.